src/lib/parse-cdn.ts

```diff
@@ -1,31 +1,48 @@
 import { embeddedWriteKey } from './embedded-write-key'
 const regex = /(https:\/\/.*)\/analytics\.js\/v1\/(?:.*?)\/(?:platform|analytics.*)?/
 
-export function getCDN(): string {
-  let cdn: string | undefined = undefined
+export class CDN {
```

Comment:
I would push back in adding more classes to Segment, classes are not tree-shakeable nor do they allow minify method name.

Instead, export functions as it was before that works with a given CDN data structure (string)